### ISSUE TITLE
Memory leak when calling MutableArrayObject.AddBlob

### ISSUE THREAD

**Issue by dan3988** [couchbase/couchbase-lite-net#1302]
### Library Version
CouchbaseLite/2.8.4 (.NET; Microsoft Windows 10.0.19042) Build/8 LiteCore/2.8.4 (7) Commit/6f01e8d5

### .NET Runtime
.Net Core 3.1

### Operating System / Device Details
Windows 10

### Log Output

```
2021-7-5 04:42:34.916+01:00 [1]| INFO)  [Database] (Startup) [1] CouchbaseLite/2.8.4 (.NET; Microsoft Windows 10.0.19042) Build/8 LiteCore/2.8.4 (7) Commit/6f01e8d5
```

### Expected behavior
When adding a large blob object that was initialized with a byte array to a MutableArrayObject, it should be quick

### Actual behavior
When adding a large blob object that was initialized with a byte array to a MutableArrayObject, it will take a long time and use an extreme amount of memory.

### Steps To Reproduce

1. Install Couchbase.Lite 2.8.4 on a console project
2. Create an empty MutableArrayObject
3. Create a blob object with a 20MB byte array
4. Call AddBlob on the array
3. This call will take over 20 seconds and use more than 2GB of memory

![image](https://user-images.githubusercontent.com/40858622/124495904-9f358700-ddb0-11eb-9107-2bb6418101ba.png)

When downgrading Couchbase.Lite to 2.7.1, this is instant.

### Reproduction Project
https://github.com/dan3988/CouchbaseLiteMemoryLeak

**Comment by Sandychuang8** [comment]
Thanks for bring this up to our attention. Will work on the fix and the fix will be in our 3.0 release.

**Comment by Sandychuang8** [comment]
Hi there, new release 2.8.6 contains the fix.
